java/fury-core/src/main/java/org/apache/fury/builder/CompatibleCodecBuilder.java

```diff
@@ -341,9 +341,13 @@ private Expression writeEmbedTypeFieldValue(
       Expression bean, Expression buffer, FieldInfo fieldInfo) {
     Descriptor descriptor = createDescriptor(fieldInfo);
     walkPath.add(descriptor.getDeclaringClass() + descriptor.getName());
+    boolean nullable = false;
+    if (!descriptor.getTypeRef().isPrimitive()) {
```

Comment:
Could we add a nullable method in Descriptor?

Reply:
I think we can have a nullable/trackingRef field and getter in Descriptor.

We can also have a Descriptor which can build Descriptor with different field values easily.

```java
Descriptor desc = ...;
new DescriptorBuilder(desc).nullable(true).trackingRef(false).build();
desc.builder().nullable(true).trackingRef(false).build();
```